stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -267,6 +268,12 @@ struct ConformanceState {
   ConcurrentMap<ConformanceCacheEntry> Cache;
   ConcurrentReadableArray<ConformanceSection> SectionsToScan;
   
+  llvm::DenseMap<const ProtocolDescriptor *,
+                 llvm::SmallDenseSet<const ProtocolConformanceDescriptor *, 1>>
```

Comment:
By construction, this cache won't have any duplicates in it, so you should be able to use a `TinyPtrVector` here instead of a `SmallDenseSet`.